tests/scripts/all.sh

```diff
@@ -34,13 +34,26 @@ MEMORY=0
 SHORT=0
 FORCE=0
 
+OPENSSL="openssl"
+OPENSSL_LEGACY="$OPENSSL"
```

Comment:
I'm not sure there's much point defaulting to existing OpenSSL as all modern versions (which you should have installed) won't support the older features we're trying to test. It's probably best to fail - because no legacy versions have been specified - and then provide an override option of '--no-legacy-tests' to skip these tests.